File: Lab-1/Task-2/JavaToScalaCall/src/com/bibletoon/Main.scala
package com.bibletoon

import scala.util.chaining.scalaUtilChainingOps;

sealed trait Identifier;
case class Username(firstName : String, secondName: String) extends Identifier;
case class Email(value : String) extends Identifier;

object Scala {
  def checkUser(id: Identifier) = id match {
    case Username(firstName, secondName) => Left(secondName+" "+firstName)
    case Email(email) => Right(email.reverse)
  }

  def main(args: Array[String]) = {
    checkUser(Username("aboba", "abobievich")) match {
      case Left(value) => value.pipe(println).pipe(println)
      case Right(value) => println(value)
    }
  }
}

File: Lab-1/Task-2/JavaToScalaCall/src/com/bibletoon/Main.java
package com.bibletoon;

import scala.util.Left;
import scala.util.Right;

public class Main {

    public static void main(String[] args) {
        Object s =  Scala.checkUser(new Username("Aboba", "Abobievich"));
        System.out.println(s.getClass().getName());
        var t = s instanceof Right ? ((Right) s).value() : ((Left) s).value();
        System.out.println(t);
    }
}
